tests/test_iqm_sampler.py

```diff
@@ -73,7 +73,26 @@ def test_run_sweep_executes_circuit_with_calibration_set_id(base_url, circuit, i
     calibration_set_id = uuid.uuid4()
     sampler = IQMSampler(base_url, Adonis(), calibration_set_id=calibration_set_id)
     run_result = RunResult(status=Status.READY, measurements=[{'some stuff': [[0], [1]]}], metadata=iqm_metadata)
-    when(client).submit_circuits(ANY, calibration_set_id=calibration_set_id, shots=ANY).thenReturn(run_id)
+    when(client).submit_circuits(
+        ANY, calibration_set_id=calibration_set_id, shots=ANY, circuit_duration_check=ANY
+    ).thenReturn(run_id)
+    when(client).wait_for_results(run_id).thenReturn(run_result)
+
+    sampler._client = client
+    results = sampler.run_sweep(circuit, None, repetitions=2)
+    assert isinstance(results[0], cirq.Result)
+
+
+@pytest.mark.usefixtures('unstub')
+def test_run_sweep_executes_circuit_with_duration_check_disabled(base_url, circuit, iqm_metadata):
+    client = mock(IQMClient)
+    run_id = uuid.uuid4()
+    calibration_set_id = uuid.uuid4()
+    sampler = IQMSampler(base_url, Adonis(), calibration_set_id=calibration_set_id, circuit_duration_check=False)
+    run_result = RunResult(status=Status.READY, measurements=[{'some stuff': [[0], [1]]}], metadata=iqm_metadata)
+    when(client).submit_circuits(
+        ANY, calibration_set_id=calibration_set_id, shots=ANY, circuit_duration_check=False
+    ).thenReturn(run_id)
```

Comment:
```suggestion
    sampler = IQMSampler(base_url, Adonis(), circuit_duration_check=False)
    run_result = RunResult(status=Status.READY, measurements=[{'some stuff': [[0], [1]]}], metadata=iqm_metadata)
    when(client).submit_circuits(ANY, calibration_set_id=ANY, shots=ANY, circuit_duration_check=False).thenReturn(
        run_id
    )
```
Not important, but slightly simpler since this doesn't need to test `calibration_set_id`.